File: 9. Palindrome Number/9. Palindrome Number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if(x<0||(x%10==0&&x>=10)){
        return false
    }
    let sum=0;
    while(x>sum){
        sum=sum*10+x%10;
        x=Math.floor(x/10);
    }
    return x==sum || x==Math.floor(sum/10)
};

File: 9. Palindrome Number/9. Palindrome Number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x<0 or (x%10==0 and x>=10):
            return False

        sum = 0
        while x>sum:
            sum=sum*10+x%10
            x=x/10
        return x == sum or x==sum/10
